snapshot/manifest.go

```diff
@@ -18,9 +18,10 @@ type Manifest struct {
 	ID     manifest.ID `json:"id"`
 	Source SourceInfo  `json:"source"`
 
-	Description string          `json:"description"`
-	StartTime   fs.UTCTimestamp `json:"startTime"`
-	EndTime     fs.UTCTimestamp `json:"endTime"`
+	Description string           `json:"description"`
+	StartTime   fs.UTCTimestamp  `json:"startTime"`
+	EndTime     fs.UTCTimestamp  `json:"endTime"`
+	Progress    map[string]int64 `json:"progress"`
```

Comment:
I wasn't sure how exactly to expose `progress` here. Would it be better to introduce a new struct? Can I somehow directly use the `cliProgress` struct? Is using a map acceptable here?

Reply:
doesn't `Stats` below have what you need?

Reply:
`Stats` doesn't contain an uploadedBytes field. The only byte fields it has are `TotalFileSize` and `ExcludedTotalFileSize`. What would be the best option to continue? For example, I could try to add the new fields `UploadedTotalFileSize`  and ` CachedTotalFileSize`   to `Stats`. Or I could keep the map that I introduced. Or I could create a new Stats struct that is only used to report the data in the json file. Thoughts :)?

Reply:
Let's do Stats. I would prefer to be intentional and only add 1 or 2 field as needed, I know some folks manage millions of manifests and bloat can already be a problem.